src/main/java/com/brandwatch/internship/fetchandstoretweets/services/FetchAndStoreService.java

```diff
@@ -0,0 +1,67 @@
+package com.brandwatch.internship.fetchandstoretweets.services;
+
+import com.brandwatch.internship.fetchandstoretweets.entities.Mention;
+import com.brandwatch.internship.fetchandstoretweets.entities.Query;
+import com.brandwatch.internship.fetchandstoretweets.repositories.MentionsRepository;
+import com.brandwatch.internship.fetchandstoretweets.repositories.QueryRepository;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.social.twitter.api.SearchResults;
+import org.springframework.social.twitter.api.Tweet;
+import org.springframework.social.twitter.api.Twitter;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+@Service
+public class FetchAndStoreService {
+
+    private Twitter twitter;
+
+    private MentionsRepository mentionsRepository;
+    private QueryRepository queryRepository;
+
+    private int counter;
+    private String[] searchStrings = {"#maven", "#spring", "#intellij_idea", "#kafka"};
+
+    @Autowired
+    public FetchAndStoreService(Twitter twitter, MentionsRepository mentionsRepository,
+                                QueryRepository queryRepository) {
+
+        this.twitter = twitter;
+        this.mentionsRepository = mentionsRepository;
+        this.queryRepository = queryRepository;
+
+        this.mentionsRepository.deleteAll();
+        this.queryRepository.deleteAll();
+
+    }
+
+    @Scheduled(fixedRate = 5000)
+    public void run() {
+
+        Query query = new Query(counter + 1, searchStrings[counter++ % 4]);
+
+        SearchResults searchResults= twitter.searchOperations().search(query.getSearchString());
+
+        List<Tweet> tweets = searchResults.getTweets();
+        List<Mention> mentions = Mention.tweetsToMentions(tweets, query.getId());
+
+        mentionsRepository.saveAll(mentions);
+        queryRepository.save(query);
```

Comment:
So every 5 seconds you save a new query? Why is that?

If you want to use query objects and query matching, the idea is that you store a query (or queries) manually via the api, and then just use them to regularly fetch data from twitter and/or match them to create mentions.

So you can either:
 - Just fetch _some_ data from twitter via the api, and then match against your queries in the code, and create mention objects
 - Actually use the query strings to query from twitter and thus create the Mention objects directly

If the Query-Mention relation is not entirely clear we can have a chat and I can explain again :)

Reply:
I think I got it now but yes I would prefer to have a chat just to make sure :-)